trino-connector/src/main/java/org/apache/gravitino/trino/connector/system/table/GravitinoSystemTableCatalog.java

```diff
@@ -56,21 +60,35 @@ public GravitinoSystemTableCatalog(CatalogConnectorManager catalogConnectorManag
 
   @Override
   public Page loadPageData() {
-    List<GravitinoCatalog> catalogs = catalogConnectorManager.getCatalogs();
-    int size = catalogs.size();
+    List<GravitinoCatalog> gravitinoCatalogs = new ArrayList<>();
+    catalogConnectorManager
+        .getUsedMetalakes()
+        .forEach(
+            (metalakeName) -> {
+              GravitinoMetalake metalake = catalogConnectorManager.getMetalake(metalakeName);
+              String[] catalogNames = metalake.listCatalogs();
+              for (String catalogName : catalogNames) {
+                Catalog catalog = metalake.loadCatalog(catalogName);
+                if (catalog.type() == Catalog.Type.RELATIONAL) {
+                  gravitinoCatalogs.add(new GravitinoCatalog(metalakeName, catalog));
+                }
+              }
+            });
+    int size = gravitinoCatalogs.size();
 
```

Comment:
In the Trino worker node. catalogs can not load to the `catalogConnectorManager`, we can retrieve catalogs from the server